lib/gem_content.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class GemContent
+  CONTENT_ROOT_FORMAT = "gems/%{gem}/contents/"
+  CONTENT_KEY_FORMAT = "gems/%{gem}/contents/%{hash}"
+  HASHES_ROOT_FORMAT = "gems/%{gem}/checksums/"
+  HASHES_KEY_FORMAT = "gems/%{gem}/checksums/%{version}.sha256"
+  PATH_ROOT_FORMAT = "gems/%{gem}/paths/%{version}/"
+  PATH_KEY_FORMAT = "gems/%{gem}/paths/%{version}/%{path}"
+  SPEC_KEY_FORMAT = "gems/%{gem}/specs/%{gem}-%{version}.gemspec"
+
+  class << self
+    def fs
+      @fs || RubygemFs.instance
```

Comment:
should this use a different S3 bucket from where we store the full gems?

Reply:
I think that would make it easier to attribute costs, potentially set different versioning policies, etc

Reply:
Yes, added.